tests/std/tests/P0896R4_views_common/test.cpp

```diff
@@ -0,0 +1,383 @@
+            static_assert(noexcept(rng | views::common) == is_noexcept);
+        }
+
+        // ... with const lvalue argument
+        if constexpr (is_view && copyable<V>) {
+            constexpr bool is_noexcept = is_nothrow_copy_constructible_v<V>;
+
+            static_assert(same_as<decltype(views::common(as_const(rng))), V>);
+            static_assert(noexcept(views::common(as_const(rng))) == is_noexcept);
+
+            static_assert(same_as<decltype(as_const(rng) | views::common), V>);
+            static_assert(noexcept(as_const(rng) | views::common) == is_noexcept);
+        } else if constexpr (!is_view) {
+            using RC                   = views::all_t<const remove_reference_t<Rng>&>;
+            constexpr bool is_noexcept = is_nothrow_constructible_v<RC, const remove_reference_t<Rng>&>;
+
+            static_assert(same_as<decltype(views::common(as_const(rng))), RC>);
+            static_assert(noexcept(views::common(as_const(rng))) == is_noexcept);
+
+            static_assert(same_as<decltype(as_const(rng) | views::common), RC>);
+            static_assert(noexcept(as_const(rng) | views::common) == is_noexcept);
+        }
+        // ... with rvalue argument
+        static_assert(CanViewCommon<remove_reference_t<Rng>> == is_view || enable_borrowed_range<remove_cvref_t<Rng>>);
+        if constexpr (is_view) {
+            constexpr bool is_noexcept = is_nothrow_move_constructible_v<V>;
+            static_assert(same_as<decltype(views::common(move(rng))), V>);
+            static_assert(noexcept(views::common(move(rng))) == is_noexcept);
+
+            static_assert(same_as<decltype(move(rng) | views::common), V>);
+            static_assert(noexcept(move(rng) | views::common) == is_noexcept);
+        } else if constexpr (enable_borrowed_range<Rng>) {
+            using S                    = decltype(ranges::subrange{declval<Rng>()});
+            constexpr bool is_noexcept = noexcept(S{declval<Rng>()});
+
+            static_assert(same_as<decltype(views::common(move(rng))), V>);
+            static_assert(noexcept(views::common(move(rng))) == is_noexcept);
+
+            static_assert(same_as<decltype(move(rng) | views::common), V>);
+            static_assert(noexcept(move(rng) | views::common) == is_noexcept);
+        }
+
+        // ... with const rvalue argument
+        static_assert(CanViewCommon<const remove_reference_t<Rng>> == (is_view && copyable<V>)
+                      || (!is_view && enable_borrowed_range<remove_cvref_t<Rng>>) );
+        if constexpr (is_view && copyable<V>) {
+            constexpr bool is_noexcept = is_nothrow_copy_constructible_v<V>;
+
+            static_assert(same_as<decltype(views::common(move(as_const(rng)))), V>);
+            static_assert(noexcept(views::common(as_const(rng))) == is_noexcept);
+
+            static_assert(same_as<decltype(move(as_const(rng)) | views::common), V>);
+            static_assert(noexcept(as_const(rng) | views::common) == is_noexcept);

```

Comment:
The `noexcept` tests here have forgotten to `move`.

Reply:
Done